core/vdbe/datetime.rs

```diff
@@ -799,38 +685,36 @@ mod tests {
                 OwnedValue::Text(Rc::new("22:30:45-05:00".to_string())),
                 next_time_str,
             ),
-            (
-                OwnedValue::Text(Rc::new("22:30:45Z".to_string())),
-                test_time_str,
-            ),
-            // Format 10: HH:MM:SS.SSS
-            (
-                OwnedValue::Text(Rc::new("22:30:45.123".to_string())),
-                test_time_str,
-            ),
-            (
-                OwnedValue::Text(Rc::new("22:30:45.123+02:00".to_string())),
-                prev_time_str,
-            ),
-            (
-                OwnedValue::Text(Rc::new("22:30:45.123-05:00".to_string())),
-                next_time_str,
-            ),
-            (
-                OwnedValue::Text(Rc::new("22:30:45.123Z".to_string())),
-                test_time_str,
-            ),
-            // Test Format 11: 'now'
-            (OwnedValue::Text(Rc::new("now".to_string())), &now),
-            // Format 12: DDDDDDDDDD (Julian date as float or integer)
-            (OwnedValue::Float(2460082.1), "14:24:00"),
-            (OwnedValue::Integer(2460082), "12:00:00"),
         ];
 
         for (input, expected) in test_cases {
+            // Add a small sleep to avoid race conditions
+            thread::sleep(Duration::from_millis(50));
```

Comment:
I don't think time sleeps are a thing we want to do to test things, this introduces undeterministic behaviour, what is the reasoning behind it?

Reply:
The reasoning behind the sleep, is the behavior of the test itself, because of a race condition, we need to sync the execution of the test. Do you have another path of thinking, to develop this resolution?